### PR TITLE
body rendering

### PR DESCRIPTION
**Pull request by MrMuMu**
# Description

This PR makes it easier to use the render function of `BodyComponent` with the flame coordinate system

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] My PR does not decrease the code coverage, or I have __a very special case__ and explained on the PR description why this PR decreases the coverage.
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flame Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [x] Yes, this is a breaking change (please indicate a breaking change in `CHANGELOG.md`).
- [ ] No, this is *not* a breaking change.

## Related Issues

*Replace this paragraph with a list of issues related to this PR from the [issue database]. Indicate, which of these issues are resolved or fixed by this PR.*

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md

### CODE REVIEW

**Comment by MrMuMu** [comment]
I believe this is a breaking change since changes the behavior of render function of the body component

File: packages/flame_forge2d/lib/body_component.dart [hunk 1]
```diff
@@ -58,22 +58,18 @@ abstract class BodyComponent<T extends Forge2DGame> extends BaseComponent
...
+      _transform.rotateZ(-angle);
       _lastAngle = angle;
     }
-    canvas.transform(_transform.storage);
-  }
 
-  @override
-  void postRender(Canvas canvas) {
-    canvas.transform(_flipYTransform.storage);
+    canvas.transform(_transform.storage);
```
Comment (spydon) [inline packages/flame_forge2d/lib/body_component.dart:66]: This should be removed I think?

Comment (MrMuMu) [reply packages/flame_forge2d/lib/body_component.dart:66]: It is removed actually, git is messing with you :D check source

Comment (spydon) [reply packages/flame_forge2d/lib/body_component.dart:66]: There is only one commit, and it exists in that commit?

Comment (MrMuMu) [reply packages/flame_forge2d/lib/body_component.dart:66]: image

this is how it looks

File: packages/flame_forge2d/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # CHANGELOG
 
+## [0.8.1-releasecandidate.14]
```
Comment (spydon) [inline packages/flame_forge2d/CHANGELOG.md:3]: This should just say `[Next]`

File: packages/flame_forge2d/lib/body_component.dart (hunk 1 above)
Comment (spydon) [reply packages/flame_forge2d/lib/body_component.dart:66]: Weeeird, never seen github do that before.

File: packages/flame_forge2d/CHANGELOG.md (hunk 2 above)
Comment (MrMuMu) [reply packages/flame_forge2d/CHANGELOG.md]: done

File: packages/flame_forge2d/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,8 @@
 # CHANGELOG
 
+## [Next]
+ - Behavior of BodyComponent render has changed so that it can work with flame coordinate system
```
Comment (wolfenrain) [inline packages/flame_forge2d/CHANGELOG.md:4]:
```suggestion
 - The rendering of `BodyComponent` is now inline with the Flame coordinate system
```

File: packages/flame_forge2d/lib/body_component.dart (hunk 1 above)
Comment (spydon) [reply packages/flame_forge2d/lib/body_component.dart:66]: Aaah, now I understand what the diff is showing! I must have been very tired when reviewing this yesterday. Missed that it was red lines all the way into the render method.

**Review by spydon — APPROVED** [review]

**Review by wolfenrain — APPROVED** [review]

**Review by erickzanardo — APPROVED** [review]

**Pull request merged**